File: AccessoriesEcommerce.API/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace AccessoriesEcommerce.API.Models;

public class Order
{
    public int Id { get; set; }
    
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    
    [Required]
    public string OrderNumber { get; set; } = string.Empty;
    
    [Required]
    public string Status { get; set; } = "Pending"; // Pending, Shipped, Delivered
    
    [Required]
    public decimal TotalAmount { get; set; }
    
    [Required]
    public string ShippingAddress { get; set; } = string.Empty;
    
    [Required]
    public string ShippingCity { get; set; } = string.Empty;
    
    [Required]
    public string ShippingPostalCode { get; set; } = string.Empty;
    
    [Required]
    public string ShippingCountry { get; set; } = string.Empty;
    
    [Required]
    public string CustomerName { get; set; } = string.Empty;
    
    [Required]
    [Phone]
    public string CustomerPhone { get; set; } = string.Empty;
    
    public string? Notes { get; set; }
    
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    
    public DateTime? ShippedDate { get; set; }
    
    public DateTime? DeliveredDate { get; set; }
    
    // Navigation properties
    public List<OrderItem> OrderItems { get; set; } = new();
}

File: frontend/src/types/order.ts
export interface OrderItemDto {
  id: number;
  productId: number;
  productName: string;
  productImage: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface OrderDto {
  id: number;
  orderNumber: string;
  status: string;
  totalAmount: number;
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  customerName: string;
  customerPhone: string;
  notes?: string;
  orderDate: string;
  shippedDate?: string;
  deliveredDate?: string;
  orderItems: OrderItemDto[];
}

export interface CreateOrderRequest {
  shippingAddress: string;
  shippingCity: string;
  shippingPostalCode: string;
  shippingCountry: string;
  customerName: string;
  customerPhone: string;
  notes?: string;
  items: OrderItemRequest[];
}

export interface OrderItemRequest {
  productId: number;
  quantity: number;
}
